nengo_gui/static/datastore.js

```diff
@@ -105,6 +98,7 @@ Nengo.DataStore.prototype.get_shown_data = function() {
 
     /* find the corresponding index values */
     var index = 0;
+    // Wouldn't a binary search be better?
```

Comment:
> Probably, except if we make only a few loops until the value is found (I don't know this part of the code enough to judge how many loops are expected on average.)

Often a lot...  In the normal use case there are ~4000 items in the list and the item it's looking for is # ~3500.  

> Also depends on how often this code is called (if it's only called a few times per execution, it might not be worth to improve the efficiency, but this seems to be called on a regular basis?)

It's called once per frame per graph.  So also a lot.  :)

> How regular are the values in this.times? If we expect a constant increase in most cases (with maybe a few dropped times), we could use a heuristic to estimate the index which might be even better than binary search.

99% of the time they are completely regular, since we get data every dt.  The only times that won't be the case will be when running on the spinnaker backend and other real-time hardware systems.

So yeah, some serious optimizing of that should be done.  But that should be a separate PR from this.